sqlglot/transforms.py

```diff
@@ -1003,53 +1007,37 @@ def _inline_inherited_window(window: exp.Expression) -> None:
 
 def inherit_struct_field_names(expression: exp.Expression) -> exp.Expression:
     """
-    Inherit struct field names from the first struct in an array to subsequent unnamed structs.
+    Inherit field names from the first struct in an array.
 
-    BigQuery (and some other dialects) allow shorthand where the first STRUCT in an array
-    defines field names and subsequent STRUCTs inherit them:
+    BigQuery supports implicitly inheriting names from the first STRUCT in an array:
 
     Example:
         ARRAY[
           STRUCT('Alice' AS name, 85 AS score),  -- defines names
-          STRUCT('Bob', 92),                      -- inherits names
-          STRUCT('Diana', 95)                     -- inherits names
+          STRUCT('Bob', 92),                     -- inherits names
+          STRUCT('Diana', 95)                    -- inherits names
         ]
 
-    This transform makes the field names explicit on all structs by adding PropertyEQ nodes.
+    This transformation makes the field names explicit on all structs by adding
+    PropertyEQ nodes, in order to facilitate transpilation to other dialects.
 
     Args:
         expression: The expression tree to transform
 
     Returns:
         The modified expression with field names inherited in all structs
     """
-    for array in expression.find_all(exp.Array):
-        if not array.expressions:
-            continue
-
-        # Check if first element is a Struct with field names
-        first_item = array.expressions[0]
-        if not isinstance(first_item, exp.Struct):
-            continue
-
-        # Get field names from first struct (PropertyEQ nodes)
-        property_eqs = [e for e in first_item.expressions if isinstance(e, exp.PropertyEQ)]
-        if not property_eqs:
-            continue
-
-        field_names = [pe.name for pe in property_eqs]
+    if (
+        isinstance(expression, exp.Array)
+        and expression.args.get("struct_name_inheritance")
```

Comment:
Added this to ensure we only mimic struct name inheritance when the source dialect supports it